python-package/xgboost/spark/data.py

```diff
@@ -208,10 +208,14 @@ def create_dmatrix_from_partitions(  # pylint: disable=too-many-arguments
 
     def append_m(part: pd.DataFrame, name: str, is_valid: bool) -> None:
         nonlocal n_features
-        if name in part.columns and part[name].shape[0] > 0:
```

Comment:
why get rid of the check "part[name].shape[0] > 0"

Reply:
The empty data tests are passing.

Reply:
what? really? I will check it, please hold on.

Reply:
We must add check `part[feature_cols].shape[0] > 0`, or else, stack_series will throw exception. The latest change is ok.